pkg/endpoint/endpoint.go

```diff
@@ -361,14 +397,122 @@ func (e *Endpoint) WaitForProxyCompletions() error {
 	return nil
 }
 
+// runK8sCiliumEndpointSync starts a controller that syncronises the endpoint
+// to the corresponding k8s CiliumEndpoint CRD
+// ep.Mutex must be help for the setup portion of this function, but not the
+// controller.
+// CiliumEndpoint objects follow a nodename-cep-endpointID scheme
+func (e *Endpoint) runK8sCiliumEndpointSync() {
+	var (
+		endpointID     = e.ID
+		controllerName = fmt.Sprintf("sync-to-k8s-ciliumendpoint (%v)", endpointID)
+		epName         = fmt.Sprintf("%v-cep-%v", node.GetName(), endpointID)
+		scopedLog      = e.getLogger().WithField("controller", controllerName)
+	)
+
+	if !k8s.IsEnabled() {
+		scopedLog.Debug("Not starting controller because k8s is disabled")
+		return
+	}
+
+	// The health endpoint is created outside of k8s and shouldn't create a CEP
+	// for it. It has a specific label on it and we do not start the controller
+	// in that case.
+	healthLbl := labels.NewLabel(labels.IDNameHealth, "", labels.LabelSourceReserved)
```

Comment:
What's the rationale to hide it?

Reply:
We can consider showing it in the `kube-system` namespace

Reply:
I did have that to start (kube-system). It was hacky because we were forcing a k8s namespace into it but not any other k8s related field. I also worried that it would confuse people to see endpoints without any corresponding pods.
shrug. It's up to you, I don't really have a strong opinion.

Reply:
I'm ok as-is

Reply:
oops, sorry, I put it back in yesterday. I can remove it, but I'll assume it doesn't matter that much and avoid the added churn.